Day-2/problem-3.go

```diff
@@ -8,27 +8,29 @@ import (
 type Account struct {
 	balance uint
 	mu      sync.Mutex
-	wg      sync.WaitGroup
 }
 
-func (account *Account) deposit(amount uint) {
-	defer account.wg.Done()
+func (account *Account) deposit(wg *sync.WaitGroup, amount uint) {
+	defer wg.Done()
 	account.mu.Lock()
 	account.balance += amount
 	account.getBalance()
 	account.mu.Unlock()
 }
 
-func (account *Account) withdraw(amount uint) {
-	defer account.wg.Done()
+func (account *Account) withdraw(wg *sync.WaitGroup, amount uint) {
+	defer func() {
+		account.mu.Unlock()
+		wg.Done()
+	}()
 	account.mu.Lock()
 	if account.balance < amount {
 		fmt.Printf("Withdrawl amount %d exceeds balance %d \n", amount, account.balance)
```

Comment:
Could use a error channel to indicate error and print a error message.

Reply:
Okay